pkgs/build-support/rust/build-rust-crate/default.nix

```diff
@@ -250,7 +250,7 @@ stdenv.mkDerivation (rec {
       depsMetadata = lib.foldl' (str: dep: str + dep.metadata) "" (dependencies ++ buildDependencies);
       hashedMetadata = builtins.hashString "sha256"
         (crateName + "-" + crateVersion + "___" + toString (mkRustcFeatureArgs crateFeatures) +
-          "___" + depsMetadata);
+          "___" + depsMetadata + "___" + stdenv.hostPlatform.config);
```

Comment:
Can we do something with `toRustTarget` instead? There are some cases where that target is different than `hostPlatform.config` and I think it would be good to follow it in those cases.